File: 01_ProgrammingBasics/_06_NestedLoopsExercise/_01_NumberPyramid.java
package _01_ProgrammingBasics._06_NestedLoopsExercise;

import java.util.Scanner;

public class _01_NumberPyramid {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        int num = Integer.parseInt(scan.nextLine());

        // Add variables
        int current = 1;
        boolean isBigger = false;
        
        // Print the number pyramid
        for (int rows = 1; rows <= num; rows++) {
            for (int cols = 1; cols <= rows; cols++) {
                if (current > num) {
                    isBigger = true;
                    break;
                }
                
                System.out.print(current + " ");
                current++;
            }
            
            if (isBigger) {
                break;
            }
            
            System.out.println();
        }

        scan.close();
    }
}

File: 23_C#Basics/_06_NestedLoopsExercise/_01_NumberPyramid/_01_NumberPyramid.cs
using System;

public class _01_NumberPyramid
{
    public static void Main(string[] args)
    {
        // Read input value
        int num = int.Parse(Console.ReadLine());

        // Add variables
        int current = 1;
        bool isBigger = false;

        // Print the number pyramid
        for (int rows = 1; rows <= num; rows++)
        {
            for (int cols = 1; cols <= rows; cols++)
            {
                if (current > num)
                {
                    isBigger = true;
                    break;
                }

                Console.Write(current + " ");
                current++;
            }

            if (isBigger)
            {
                break;
            }

            Console.WriteLine();
        }
    }
}
